### DISCUSSION TITLE
Convergence difficulties met in LCAO scf calculation with Broyden and Kerker charge mixing method

### DISCUSSION THREAD

**Discussion by hongriTianqi** [deepmodeling/abacus-develop#2296] [General]
## Describe the Bug
Convergence difficulties are met in LCAO scf calculation with Broyden and Kerker charge mixing methods, but not in the corresponding PW calculation.

## To Reproduce
1. cat 206_NO_BD_chargemixing/OUT.autotest/running_scf.log
2. cat 206_NO_KK_chargemixing/OUT.autotest/running_scf.log
3. cat 106_PW_BD_chargemixing/OUT.autotest/running_scf.log
4. cat 106_PW_KK
[chargemixing.tar.gz](https://github.com/deepmodeling/abacus-develop/files/7659379/chargemixing.tar.gz)
_chargemixing/OUT.autotest/running_scf.log

**Comment by mohanchen** [comment]
@YuLiu98 Could you take a look at these examples and propose your opinions?

**Comment by hongriTianqi** [comment]
This issue has not been addressed, so it should be kept open. @Unboundwill @dyzheng @YuLiu98 @mohanchen

**Comment by caic99** [comment]
@Qianruipku @mohanchen Would you look into this issue?

**Comment by Qianruipku** [comment]
Maybe it is associated with issue #484. scf aims to find the minimum of "dr2" when it is thought to be converged. However, "dr2" is differently defined for lcao and pw. Thus, we must use different mixing methods for lcao and pw. In fact, "pulay" and "broyden" are both the pulay method.  The former is for lcao while the latter is for pw.

**Comment by wenfei-li** [comment]
We might need density matrix mixing. But we’ll decide that later since it will involve lots of refactor
